File: csharp/calculator-conundrum/CalculatorConundrum.cs
public static class SimpleCalculator
{
    public static string Calculate(int operand1, int operand2, string? operation)
    {
        if (operation == null)
        {
            throw new ArgumentNullException("Operation cannot be null");
        }

        if (operation == string.Empty)
        {
            throw new ArgumentException("Operation cannot be empty");
        }

        if (operation == "/" && operand2 == 0)
        {
            return "Division by zero is not allowed.";
        }

        var result = operation switch
        {
          "+" => SimpleOperation.Addition(operand1, operand2),
          "/" => SimpleOperation.Division(operand1, operand2),
          "*" => SimpleOperation.Multiplication(operand1, operand2),
          _ => throw new ArgumentOutOfRangeException("Invalid operation")  
        };

        return $"{operand1} {operation} {operand2} = {result}";
    }
}

File: java/calculator-conundrum/src/main/java/CalculatorConundrum.java
class CalculatorConundrum {
    public String calculate(int operand1, int operand2, String operation) {
        if (operation == null) {
            throw new IllegalArgumentException("Operation cannot be null");
        }

        if (operation.isEmpty()) {
            throw new IllegalArgumentException("Operation cannot be empty");
        }

        int result = switch (operation) {
            case "+" -> operand1 + operand2;
            case "*" -> operand1 * operand2;
            case "/" -> {
                try {
                    yield operand1 / operand2;
                } catch (Throwable e) {
                    throw new IllegalOperationException("Division by zero is not allowed", e);
                }
            }
            default -> throw new IllegalOperationException("Operation '%s' does not exist".formatted(operation));
        };

        return "%d %s %d = %d".formatted(operand1, operation, operand2, result);
    }
}
